ekncontent/ekncontent/eknc-domain.c

```diff
@@ -299,9 +307,14 @@ eknc_domain_initable_init (GInitable *initable,
   g_autoptr(GFile) bundle_subscriptions_dir = g_file_get_child (self->content_dir, "com.endlessm.subscriptions");
   g_autoptr(GFile) bundle_dir = g_file_get_child (bundle_subscriptions_dir, self->subscription_id);
 
-  if (!g_file_query_exists (subscription_dir, cancellable) &&
-      !g_file_make_directory_with_parents (subscription_dir, cancellable, error))
-      return FALSE;
+  g_file_make_directory_with_parents (subscription_dir, cancellable, error);
+  if (*error)
+    {
+      if (g_error_matches (*error, G_IO_ERROR, G_IO_ERROR_EXISTS))
```

Comment:
It feels a little weird to be setting and unsetting the error passed in from the invoker here over the course of initialization.

The other option would be to use a local error and call g_propogate_error in all the early returns. That would be a little more verbose, I couldn't think of an actual practical reason where setting then unsetting the passed in error would bite us. But maybe makes sense to switch.